resotoclient/__init__.py

```diff
@@ -45,16 +55,82 @@ def __init__(
         psk: Optional[str],
         custom_ca_cert_path: Optional[str] = None,
         verify: bool = True,
+        renew_before: timedelta = timedelta(days=1),
     ):
-        self.base_url = url
+        self.resotocore_url = url
         self.psk = psk
-        self.ca_cert_path = None
-        if custom_ca_cert_path:
-            self.ca_cert_path = custom_ca_cert_path
-        elif url.startswith("https"):
-            self.ca_cert_path = ca.load_ca_cert(resotocore_uri=url, psk=psk)
-        self.session_id = rnd_str()
+        self.custom_ca_cert_path = custom_ca_cert_path
         self.verify = verify
+        self.__ca_cert = None
+        self.__tempdir = TemporaryDirectory(prefix="resoto-cert-")
+        self.__ca_cert_path = f"{self.__tempdir.name}/ca.crt"
+        self.session_id = rnd_str()
+        self.renew_before = renew_before
+        self.__watcher = Thread(
+            target=self.__certificates_watcher, name="certificates_watcher"
+        )
+        self.__load_lock = Lock()
+        self.__loaded = Event()
+        self.__exit = Condition()
```

Comment:
Quite some stuff and multiple responsibilities.
Maybe it is better to put the ca/refresh handling to a separate class that is used here?
That way you can separate the different responsibilities.